### PR TITLE
WRP-4228: Add `shadowed` prop to `Panels.Header` and `Button`

### PR DESCRIPTION
**Pull request by juwonjeong**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Add prop to enable shadow effects on icons and text in ```Panels.Header```.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Use text-shadow CSS property to add shadow to text.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

The UX guide only mentioned close button. But for consistency, I also applied a shadow effect to the back button (In Panels/Header.js, line 462).

### Links
[//]: # (Related issues, references)
WRP-4228

### Comments

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ before_install:
     - sudo apt-get install build-essential libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev librsvg2-dev
 install:
     - npm config set prefer-offline false
-    - npm install -g enactjs/cli#develop
+    - npm install -g @enact/cli@5.1.1
```
Comment (MikyungKim) [inline .travis.yml:12]: I guess we don't need this change anymore.

File: Button/Button.js [hunk 2]
```diff
@@ -166,6 +166,15 @@ const ButtonBase = kind({
 		 */
 		minWidth: PropTypes.bool,
 
+		/**
+		 * Boolean controlling whether this component should add shadow to the text when button is transparent.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (MikyungKim) [inline Button/Button.js:173]: We don't specify @default false.

File: Button/Button.js [hunk 3]
```diff
@@ -166,6 +166,15 @@ const ButtonBase = kind({
 		 */
 		minWidth: PropTypes.bool,
 
+		/**
+		 * Boolean controlling whether this component should add shadow to the text when button is transparent.
```
Comment (MikyungKim) [inline Button/Button.js:170]:
```suggestion
		 * Adds shadow to the text when the button is transparent.
```

File: Button/Button.js [hunk 4]
```diff
@@ -184,6 +193,7 @@ const ButtonBase = kind({
 		iconComponent: Icon,
 		iconOnly: false,
 		iconPosition: 'before',
+		shadowed: false,
```
Comment (MikyungKim) [inline Button/Button.js:196]: 🤔 I guess we don't need this either.

File: Button/Button.module.less [hunk 5]
```diff
@@ -304,6 +304,11 @@
 			.bg {
 				opacity: @sand-button-transparent-bg-opacity;
 			}
+			&.shadowed {
```
Comment (MikyungKim) [inline Button/Button.module.less:307]:
```suggestion

			&.shadowed {
```

File: Button/Button.module.less [hunk 6]
```diff
@@ -304,6 +304,11 @@
...
+				.client {
+					text-shadow: @sand-button-text-shadow;
+				}
+			}
```
Comment (MikyungKim) [inline Button/Button.module.less:311]:
```suggestion

			&.shadowed .client {
				text-shadow: @sand-button-text-shadow;
			}
```

File: .travis.yml (hunk 1 above)
Comment (MikyungKim) [reply .travis.yml:12]: Or should we do like
- npm install -g @enact/cli
as we changed from develop branch?

File: Panels/Header.js [hunk 7]
```diff
@@ -181,6 +181,15 @@ const HeaderBase = kind({
 		 */
 		onClose: PropTypes.func,
 
+		/**
+		 * Boolean controlling whether this component should add shadow to the text.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (MikyungKim) [inline Panels/Header.js:188]: We can remove this line.

File: Panels/Header.js [hunk 8]
```diff
@@ -181,6 +181,15 @@ const HeaderBase = kind({
 		 */
 		onClose: PropTypes.func,
 
+		/**
+		 * Boolean controlling whether this component should add shadow to the text.
```
Comment (MikyungKim) [inline Panels/Header.js:185]:
```suggestion
		 * Adds shadow to the text.
```

File: Panels/Header.js [hunk 9]
```diff
@@ -327,6 +336,7 @@ const HeaderBase = kind({
 	defaultProps: {
 		marqueeOn: 'render',
 		noSubtitle: false,
+		shadowed: false,
```
Comment (MikyungKim) [inline Panels/Header.js:339]: same here.

File: Panels/tests/Header-specs.js [hunk 10]
```diff
@@ -142,4 +142,11 @@ describe('Header Specs', () => {
 		expect(closeButtonAriaLabel).toBeInTheDocument();
 		expect(slotAfter).toHaveClass('slotAfter');
 	});
+
+	test ('should set close button `shadowed` to true when `shadowed` is set to true', () => {
+		render(<Header shadowed><title>title</title></Header>);
+		const button = screen.getByRole('button');
+		const expected = 'button shadowed';
+		expect(button).toHaveClass(expected);
```
Comment (MikyungKim) [inline Panels/tests/Header-specs.js:150]:
```suggestion
		render(<Header shadowed><title>title</title></Header>);

		const button = screen.getByRole('button');
		const expected = 'button shadowed';

		expect(button).toHaveClass(expected);
```

File: internal/DateTime/tests/DateTimeDecorator-specs.js [hunk 11]
```diff
@@ -5,9 +5,9 @@ import {DateTimeDecorator} from '../';
 
 describe('DateTimeDecorator', () => {
 	test('should accept an updated JavaScript Date for its value prop', () => {
-		const Picker = DateTimeDecorator({}, function PickerBase ({locale, title, value}) {
+		const Picker = DateTimeDecorator({}, function PickerBase ({title, value}) {
```
Comment (MikyungKim) [inline internal/DateTime/tests/DateTimeDecorator-specs.js:8]: Why is this change needed?

Comment (juwonjeong) [reply internal/DateTime/tests/DateTimeDecorator-specs.js:8]: This change is added to avoid travis failure. As https://github.com/enactjs/sandstone/commit/bb662bc16a61b98746405425ed21c481a6addf82 only applied to develop branch.

File: Panels/Header.js (hunk 9 above)
Comment (juwonjeong) [reply Panels/Header.js]: It's removed.

File: Panels/Header.js (hunk 8 above)
Comment (juwonjeong) [reply Panels/Header.js]: It's fixed.

File: Panels/Header.js (hunk 7 above)
Comment (juwonjeong) [reply Panels/Header.js]: It's removed.

File: Button/Button.js (hunk 4 above)
Comment (juwonjeong) [reply Button/Button.js]: It's removed.

File: Button/Button.js (hunk 3 above)
Comment (juwonjeong) [reply Button/Button.js]: It's removed.

File: Button/Button.js (hunk 2 above)
Comment (juwonjeong) [reply Button/Button.js]: It's removed.

File: .travis.yml (hunk 1 above)
Comment (juwonjeong) [reply .travis.yml]: I changed as develop branch.

File: Button/Button.module.less [hunk 12]
```diff
@@ -304,6 +304,9 @@
 			.bg {
 				opacity: @sand-button-transparent-bg-opacity;
 			}
+			&.shadowed .client {
```
Comment (MikyungKim) [inline Button/Button.module.less:307]:
```suggestion

			&.shadowed .client {
```

File: styles/colors.less [hunk 13]
```diff
@@ -222,6 +223,10 @@
 @sand-formcheckboxitem-disabled-text-color: inherit;
 @sand-formcheckboxitem-disabled-focus-text-color: @sand-formcheckboxitem-disabled-text-color;
 
+// Header
+// ---------------------------------------
+@sand-header-text-shadow: 6px 12px 12px rgba(0, 0, 0, 0.3);
```
Comment (MikyungKim) [inline styles/colors.less:228]: I guess we could replace `rgba(0, 0, 0, 0.3)` with `@sand-shaodw-color`

File: Panels/Header.module.less [hunk 14]
```diff
@@ -92,6 +93,13 @@
 		margin: @sand-header-slotbelow-margin;
 	}
 
+	&.shadowed {
```
Comment (MikyungKim) [inline Panels/Header.module.less:96]: Instead of importing `colors.less` directly, let's move this definition into `applySkins` mixin. :D That would be better.

File: CHANGELOG.md [hunk 15]
```diff
@@ -2,6 +2,12 @@
 
 The following is a curated list of changes in the Enact sandstone module, newest changes on the top.
 
+## [unreleased]
+
+### Added
+
+- `sandstone/Panels.Header` prop `shadowed` to add shadow to text and icon
```
Comment (MikyungKim) [inline CHANGELOG.md:9]:
```suggestion
- `sandstone/Panels.Header` prop `shadowed` to add shadow to text and buttons
```

File: Button/Button.module.less (hunk 12 above)
Comment (juwonjeong) [reply Button/Button.module.less:308]: It's fixed.

File: styles/colors.less (hunk 13 above)
Comment (juwonjeong) [reply styles/colors.less]: It's replaced.

File: Panels/Header.module.less (hunk 14 above)
Comment (juwonjeong) [reply Panels/Header.module.less]: As I moved definition into applySkins, `Header.js` also fixed to apply `skinnable`.

File: CHANGELOG.md (hunk 15 above)
Comment (juwonjeong) [reply CHANGELOG.md]: It's fixed.

File: Button/Button.js [hunk 16]
```diff
@@ -166,6 +166,14 @@ const ButtonBase = kind({
 		 */
 		minWidth: PropTypes.bool,
 
+		/**
+		 * Adds shadow to the text when the button is transparent.
+		 *
```
Comment (MikyungKim) [inline Button/Button.js:171]:
```suggestion
		 * Adds shadow to the text.
		 * It is only applied when the background opacity of the button is `transparent`.
		 *
```

File: Button/Button.js [hunk 17]
```diff
@@ -166,6 +166,14 @@ const ButtonBase = kind({
...
+		 * @type {Boolean}
+		 * @private
```
[the remark this answers is not in the document]
Comment (juwonjeong) [reply Button/Button.js]: It's fixed

File: Button/tests/Button-specs.js [hunk 18]
```diff
@@ -198,6 +198,17 @@ describe('Button', () => {
 		});
 	});
 
+	describe('with shadowed', () => {
+		test('should have shadowed class', () => {
```
Comment (MikyungKim) [inline Button/tests/Button-specs.js:202]: Would you please add for a test that `should not have shadowed class when the background is not 'transparent``?
```suggestion
		test('should have shadowed class when the background is transaprent', () => {
```

**Review by MikyungKim — APPROVED** [review]
:lgtm:

**Review by 0x64 — APPROVED** [review]
